src/olympia/search/tests/test_search_ranking.py

```diff
@@ -993,22 +1005,49 @@ def test_score_boost_exact_match_in_right_language(self):
         # Note that the name returned follows the language requested.
         self._check_scenario(
             'foobar unique francais',
-            (['Foobar unique english', 1057],),
+            (['Foobar unique english', 1062],),
             lang='en-US',
         )
 
     def test_scenario_tab(self):
         self._check_scenario(
             'tab',
             (
-                ['Tabby Cat', 3730],
-                ['Tab Mix Plus', 1352],
-                ['OneTab', 1144],
-                ['Tab Center Redux', 961],
-                ['FoxyTab', 902],
-                ['Open Bookmarks in New Tab', 887],
-                ['Open image in a new tab', 715],
-                ['Open Image in New Tab', 560],
+                ['Tabby Cat', 2432],
+                ['Tab Mix Plus', 996],
+                ['OneTab', 736],
+                ['Tab Center Redux', 692],
+                ['Open Bookmarks in New Tab', 638],
+                ['FoxyTab', 551],
+                ['Open image in a new tab', 524],
+                ['Open Image in New Tab', 410],
+            ),
+        )
+
+    def test_scenario_wallet(self):
+        # Shouldn't be found: some ad-ons have the word "all" which is
+        # close but not enough.
+        self._check_scenario('wallet', ())
+
+    def test_downloadthemall(self):
+        self._check_scenario(
+            'down them all',
+            (
+                ['DownThemAll!', 3891],
+                ['All Downloader Professional', 40],
```

Comment:
interesting that this doesn't return any of the other "download" add-ons

Reply:
It can't guess that "down them all" is about "download". It matches enough words (and I assume trigrams) because "all" and "down" are present.